File: array/2016-01-20_Remove_Element.py
#!/usr/bin/env python
# encoding: utf-8

class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        count = 0
        for index in xrange(len(nums)):
            if nums[index] == val:
                continue
            else:
                nums[count] = nums[index]
                count += 1
        return count

class Solution2(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        try:
            while True:
                nums.remove(val)
        except ValueError as e:
            return len(nums)

if __name__ == "__main__":

File: array/2016-01-20_Remove_Element.cc
#include <iostream>
#include <algorithm>
#include <string>
#include <vector>
#include <map>
#include <unordered_map>

using namespace std;

class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
            int count = 0;
            for(size_t i = 0; i < nums.size(); i++) {
                if(nums[i] == val)
                    continue;
                else
                    nums[count++] = nums[i];
            }

            return count;
    }
};

class Solution2 {
public:
    int removeElement(vector<int>& nums, int val) {
        return distance(nums.begin(),remove(nums.begin(),nums.end(),val));
    }
};

int main()
{
    vector<int> data = {1,2,3,4,5,1,23,4,5,6,1,1,1,2};
    Solution2 so;
    cout << so.removeElement(data,1) << endl;
}
